packages/components/src/AtlantisThemeContext/AtlantisThemeContext.tsx

```diff
@@ -0,0 +1,129 @@
+import { darkTokens, tokens } from "@jobber/design";
+import React, {
+  PropsWithChildren,
+  createContext,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import merge from "lodash/merge";
+import {
+  AtlantisThemeContextProviderProps,
+  AtlantisThemeContextValue,
+  THEME_CHANGE_EVENT,
+  Theme,
+} from "./types";
+import { useThemeContextEventQueue } from "./useThemeContextEventQueue";
+import styles from "./AtlantisThemeContext.css";
+
+export const atlantisThemeContextDefaultValues: AtlantisThemeContextValue = {
+  theme: "light",
+  tokens: tokens,
+};
+
+const actualDarkTokens = merge({}, tokens, darkTokens);
+
+const AtlantisThemeContext = createContext(atlantisThemeContextDefaultValues);
+
+export function AtlantisThemeContextProvider({
+  children,
+  dangerouslyForceThemeForProvider,
+}: AtlantisThemeContextProviderProps) {
+  if (dangerouslyForceThemeForProvider) {
+    return (
+      <InternalStaticThemeProvider
+        dangerouslyForceThemeForProvider={dangerouslyForceThemeForProvider}
+      >
+        {children}
+      </InternalStaticThemeProvider>
+    );
+  }
+
+  return (
+    <InternalDynamicThemeProvider>{children}</InternalDynamicThemeProvider>
+  );
+}
+
+function InternalDynamicThemeProvider({ children }: PropsWithChildren) {
+  const initialTheme: Theme =
+    (globalThis.document.documentElement.dataset.theme as Theme) ?? "light";
+
+  const [internalTheme, setInternalTheme] = useState<Theme>(initialTheme);
+  const currentTokens = initialTheme === "dark" ? actualDarkTokens : tokens;
+
+  const {
+    isEmpty,
+    dequeueThemeChange,
+    themeChangeQueue,
+    handleThemeChangeEvent,
+  } = useThemeContextEventQueue();
+
+  useEffect(() => {
+    if (!globalThis.window) return;
+    globalThis.window.addEventListener(
+      THEME_CHANGE_EVENT,
+      handleThemeChangeEvent,
+    );
+
+    return () => {
+      if (!globalThis.window) return;
+      globalThis.window.removeEventListener(
+        THEME_CHANGE_EVENT,
+        handleThemeChangeEvent,
+      );
+    };
+  }, [handleThemeChangeEvent]);
+
+  useEffect(() => {
+    if (isEmpty || !globalThis.document) {
+      return;
+    }
+
+    const newTheme = dequeueThemeChange();
+    setInternalTheme(newTheme);
+  }, [themeChangeQueue, dequeueThemeChange]);
+
+  return (
+    <AtlantisThemeContext.Provider
+      value={{
+        theme: internalTheme,
+        tokens: currentTokens,
+      }}
+    >
+      {children}
+    </AtlantisThemeContext.Provider>
+  );
+}
+
+function InternalStaticThemeProvider({
+  dangerouslyForceThemeForProvider,
+  children,
+}: Required<
+  Pick<
+    AtlantisThemeContextProviderProps,
+    "dangerouslyForceThemeForProvider" | "children"
+  >
+>) {
```

Comment:
I like the pick because it was "easy" to find this when I updated the prop name in [07ea4ca](https://github.com/GetJobber/atlantis/pull/1981/commits/07ea4ca89b7c60e20b0fe74d9ada9cff17a55550). I think it also means that prop names stay consistent as they are passed down instead of "randomly" changing